hydra-access-controls/spec/unit/policy_aware_access_controls_enforcement_spec.rb

```diff
@@ -126,23 +126,27 @@ def session
   end
 
   describe "apply_gated_discovery" do
-    before do
-      allow(RoleMapper).to receive(:roles).with(user).and_return(user.roles)
-    end
     let(:governed_field) { ActiveFedora.index_field_mapper.solr_name('isGovernedBy', :symbol) }
+    let(:policy_queries) { @solr_parameters[:fq].first.split(" OR ") }
 
-    it "includes policy-aware query" do
-      # stubbing out policies_with_access because solr doesn't always return them in the same order.
-      policy_ids = (1..8).map {|n| "policies/#{n}"}
-      expect(subject).to receive(:policies_with_access).and_return(policy_ids)
-      subject.apply_gated_discovery(@solr_parameters)
-      expect(@solr_parameters[:fq].first).to include(" OR (_query_:\"{!field f=#{governed_field}}policies/1\" OR _query_:\"{!field f=#{governed_field}}policies/2\" OR _query_:\"{!field f=#{governed_field}}policies/3\" OR _query_:\"{!field f=#{governed_field}}policies/4\" OR _query_:\"{!field f=#{governed_field}}policies/5\" OR _query_:\"{!field f=#{governed_field}}policies/6\" OR _query_:\"{!field f=#{governed_field}}policies/7\" OR _query_:\"{!field f=#{governed_field}}policies/8\")")
-    end
+    before { allow(RoleMapper).to receive(:roles).with(user).and_return(user.roles) }
 
-    it "doesn't change anything if there are no clauses to add" do
-      allow(subject).to receive(:policy_clauses).and_return(nil)
-      subject.apply_gated_discovery(@solr_parameters)
-      expect(@solr_parameters[:fq].first).not_to include(" OR (_query_:\"{!field f=#{governed_field}}policies/1\" OR _query_:\"{!field f=#{governed_field}}policies/2\" OR _query_:\"{!field f=#{governed_field}}policies/3\" OR _query_:\"{!field f=#{governed_field}}policies/4\" OR _query_:\"{!field f=#{governed_field}}policies/5\" OR _query_:\"{!field f=#{governed_field}}policies/6\" OR _query_:\"{!field f=#{governed_field}}policies/7\" OR _query_:\"{!field f=#{governed_field}}policies/8\")")
+    context "when policies are included" do
+      before { subject.apply_gated_discovery(@solr_parameters) }
+      
+      it "builds a query that includes all the policies" do
+        expect(policy_queries).to include(/_query_:\"{!raw f=#{governed_field}}test-policy[1-11]\"/)
```

Comment:
I think this is basically just checking that `test-policy1` is present.  If that's the case, there's a more clear way to write that regular expression (not using the character class).